### PR TITLE
BCMOHAD-15683||Defect fix

### PR DESCRIPTION
**Pull request by ahshaik1**

### CODE REVIEW

File: TPL/force-app/main/default/classes/HealthcareCostTriggerHandler.cls [hunk 1]
```diff
@@ -1,165 +1,50 @@
...
-                System.debug('149');
-                Case c = caseMap.get(hcc.Case2__c);
-                if(hcc.RecordTypeName__c == 'Continuing_Care'){
-                    c.Continuing_Care_Total2__c = c.Continuing_Care_Total2__c - hcc.Cost__c;
-                }
-                if(hcc.RecordTypeName__c == 'Future_Care'){
-                    c.Future_Care_Total2__c = c.Future_Care_Total2__c - hcc.Cost__c;
-                }
-                caseMap.put(c.Id,c);	
+                c.id = (String)agg.get('case2__c');
```
Comment (cgijeffolsen) [inline TPL/force-app/main/default/classes/HealthcareCostTriggerHandler.cls:16]: Note to team: I approved this rollback to the original process after Ahmed worked on the solution using the original planned mechanism (aggregate queries) -- this is the core change where the default original values are set prior to processing the query results.

File: TPL/force-app/main/default/classes/HealthcareCostTriggerHandler.cls [hunk 2]
```diff
@@ -1,165 +1,50 @@
...
         for(Healthcare_Cost__c hcc : hccList){
             caseIds.add(hcc.Case2__c);
         }
-        System.debug(caseIds);
-        List<Healthcare_Cost__c> hccSumList = [select id,Total_Cost_Override__c,case2__c,RecordTypeName__c,Cost__c,Cost_Include__c,Cost_Review__c from Healthcare_Cost__c where 
-                                               case2__c IN:caseIds];
-        System.debug(hccSumList);
+        List<AggregateResult> groupedResults = [select SUM(Total_Cost_Override__c)tot,case2__c,RecordType.DeveloperName,sum(Cost__c)costSum from Healthcare_Cost__c where Case2__c IN:caseIds and ((Cost_Include__c = true and Cost_Review__c = true) or (RecordType.DeveloperName IN ('Continuing_Care','Future_Care')))    GROUP BY case2__c,RecordType.DeveloperName] ;
         List<Case> caseUpdateList = new List<Case>();
-        Decimal totalCostOverride =0;
```
Comment (cgijeffolsen) [inline TPL/force-app/main/default/classes/HealthcareCostTriggerHandler.cls:14]: This section was restored from a full looping calculation back to the original aggregrate query method.

File: TPL/force-app/main/default/classes/HealthcareCostTriggerHandler.cls [hunk 3]
```diff
@@ -1,165 +1,50 @@
...
-            caseUpdateMap.put(c1.id,c1);
-        }
-        try{
-            update caseUpdateMap.values();
-        }catch(Exception e){
-            System.debug(e.getMessage());
-        }
-        
-    } 
-    public static void updateRollupDelete(List<Healthcare_Cost__c> hccList){
```
Comment (cgijeffolsen) [inline TPL/force-app/main/default/classes/HealthcareCostTriggerHandler.cls:114]: This code is removed as handling of delete separately is not requiring when using the aggregate query approach.

**Review by cgijeffolsen — APPROVED** [review]
Approved @ahshaik1

**Review by NataliaNikishina — CHANGES_REQUESTED** [review]
@ahshaik1 The feature branch needs to be rebased. Once it is done, please update your PR. Many thanks!

**Pull request closed without merging**